File: src/main/java/com/developerphil/adbidea/ui/MyChooseModulesDialog.java
package com.developerphil.adbidea.ui;

import com.intellij.icons.AllIcons;
import com.intellij.ide.util.ChooseElementsDialog;
import com.intellij.openapi.project.Project;
import java.util.List;
import javax.swing.Icon;
import org.jetbrains.annotations.Nullable;

public class MyChooseModulesDialog extends ChooseElementsDialog<String> {

    private Icon mIcon;

    public MyChooseModulesDialog(Project project, List<? extends String> items, String title, String description, Icon icon) {
        super(project, items, title, description, false);
        mIcon = icon;
    }

    public void setSingleSelectionMode() {
        this.myChooser.setSingleSelectionMode();
    }

    @Override
    protected String getItemText(String s) {
        return s;
    }

    @Nullable
    @Override
    protected Icon getItemIcon(String s) {
        return s.equals(ModuleChooserDialogHelper.DO_NOT_SELECT_THE_DEFAULT_MODULE) ? AllIcons.Actions.Clear: mIcon;
    }
}

File: src/main/kotlin/com/developerphil/adbidea/ui/MyChooseModulesDialog.kt
package com.developerphil.adbidea.ui

import com.intellij.icons.AllIcons
import com.intellij.ide.util.ChooseElementsDialog
import com.intellij.openapi.project.Project
import javax.swing.Icon

class MyChooseModulesDialog(project: Project, items: List<String>, title: String, description: String, private val mIcon: Icon) : ChooseElementsDialog<String>(project, items,
    title, description, false) {

    fun setSingleSelectionMode() {
        this.myChooser.setSingleSelectionMode()
    }

    override fun getItemText(s: String): String {
        return s
    }

    override fun getItemIcon(s: String): Icon? {
        return if (s == ModuleChooserDialogHelper.DO_NOT_SELECT_THE_DEFAULT_MODULE) AllIcons.Actions.Close else mIcon
    }
}